guides/examples/Apache Kafka.md

```diff
@@ -0,0 +1,214 @@
+              hosts: [localhost: 9092],
+              group_id: "group_1",
+              topics: ["test"],
+            ]},
+            stages: 10
+          ],
+          processors: [
+            default: [
+              stages: 10
+            ]
+          ],
+          batchers: [
+            default: [
+              batch_size: 100,
+              batch_timeout: 200,
+              stages: 10
+            ]
+          ]
+        )
+      end
+
+      ...callbacks...
+    end
+
+For a full list of options for `BroadwayKafka.Producer`, refer to the
+official [BroadwayKafka](https://hexdocs.pm/broadway_kafka/) documentation.
+
+For general information about setting up Broadway, see `Broadway`
+module docs as well as `Broadway.start_link/2`.
+
+## Implement Broadway callbacks
+
+In order to process incoming messages, we need to implement the
+required callbacks. For the sake of simplicity, we're considering that
+all messages received from the queue are just numbers:
+
+    defmodule MyBroadway do
+      use Broadway
+
+      alias Broadway.Message
+
+      ...start_link...
+
+      @impl true
+      def handle_message(_, message, _) do
+        message
+        |> Message.update_data(fn data -> {data, String.to_integer(data) * 2} end)
+      end
+
+      @impl true
+      def handle_batch(_, messages, _, _) do
+        list = messages |> Enum.map(fn e -> e.data end)
+        IO.inspect(list, label: "Got batch")
+        messages
+      end
+    end
+
+We are not doing anything fancy here, but it should be enough for our
+purpose. First, we update the message's data individually inside
+`handle_message/3` and then we print each batch inside `handle_batch/4`.
+
+For more information, see `c:Broadway.handle_message/3` and
+`c:Broadway.handle_batch/4`.
+
+> Note: Since Broadway v0.2, batching is optional. In case you don't need to
+> group messages as batches for further processing/publishing, you can remove
+> the `:batchers` configuration along with the `handle_batch/4` callback.
+
+## Run the Broadway pipeline
+
+To run your `Broadway` pipeline, you just need to add as a child in
+a supervision tree. Most applications have a supervision tree defined
+at `lib/my_app/application.ex`. You can add Broadway as a child to a
+supervisor as follows:
+
+    children = [
+      {MyBroadway, []}
+    ]
+
+    Supervisor.start_link(children, strategy: :one_for_one)
+
+Now the Broadway pipeline should be started when your application starts.
+Also, if your Broadway has any dependency (for example, it needs to talk
+to the database), make sure that Broadway is listed *after* its dependencies
+in the supervision tree.
+
+You can now test your pipeline by entering an `iex` session:
+
+    iex -S mix
+
+If everything went fine, you should see lots of `info` log messages from the `brod`
+supervisors.
+
+### Sending messages to Kafka
+
+Finally, we can send some sample messages to Kafka using the following snippet:
```

Comment:
We should probably introduce `:brod` here and say it is also the library used by broadway_kafka to talk to kafka.